lib/auth/v2/constructStringToSign.js

```diff
@@ -35,9 +35,10 @@ function constructStringToSign(request, data, log) {
     s3cmd includes x-amz-date in amzHeaders rather
     than here in stringToSign so we have replicated that.
     */
-    const date = headers.date ? headers.date : query.Expires;
+    const date = query.Expires ? query.Expires : headers.date;
+    const combinedQueryHeaders = Object.assign(headers, query);
```

Comment:
Object.assign isn't bad by default, it just sucks if the object we are cloning is nested. Since `query` is just a single-level object, it does no more than forEach on the properties.